packages/app/src/crash-dump/containers/crash-dump/crash-dump.component.tsx

```diff
@@ -5,93 +5,149 @@
       return
     }
 
-    if (sending || downloading) {
-      setOpenSending(true)
+    if (sending || sended || failed) {
+      setOpenInfo(false)
     } else {
       setOpenInfo(true)
     }
     // AUTO DISABLED - fix me if you like :)
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [hasCrashDump])
-
-  useEffect(() => {
-    setOpenSending((sending || downloading) && hasCrashDump)
-    // AUTO DISABLED - fix me if you like :)
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [sending, downloading])
+  }, [hasCrashDump, sended, sended, failed])
 
-  const handleCloseModal = () => {
+  const handleClosCrashDumpModal = () => {
     ignoreCrashDump()
     setOpenInfo(false)
   }
 
-  const handleDownloadCrashDump = (data: SendCrashDumpPayload) => {
+  const handleDownloadCrashDump = (data: CrashDumpDto) => {
     downloadCrashDump(data)
     setOpenInfo(false)
   }
 
+  const handleCloseModal = () => {
+    resetCrashDump()
+  }
+
   if (!deviceType) {
     return <></>
   }
 
   return (
     <>
-      <CrashDumpSendingModal open={openSending} />
-      <CrashDumpModal
-        open={openInfo}
-        deviceType={deviceType}
-        onClose={handleCloseModal}
-        onSubmit={handleDownloadCrashDump}
-      />
+      {failed && (
+        <ErrorModal
+          open={failed}
+          testId={CrashDumpTestingIds.Failed}
+          title={intl.formatMessage(messages.crashDumpModalErrorTitle)}
+          subtitle={intl.formatMessage(messages.crashDumpModalErrorSubtitle)}
+          closeModal={handleCloseModal}
+        />
+      )}
+      {sending && (
+        <LoaderModal
+          open={sending}
+          testId={CrashDumpTestingIds.Loading}
+          title={intl.formatMessage(messages.crashDumpModalSendingTitle)}
+          subtitle={intl.formatMessage(messages.crashDumpModalSendingSubtitle)}
+        />
+      )}
+      {sended && (
+        <SuccessModal
+          open={sended}
+          testId={CrashDumpTestingIds.Success}
+          title={intl.formatMessage(messages.crashDumpModalSuccessTitle)}
+          subtitle={intl.formatMessage(messages.crashDumpModalSuccessSubtitle)}
+          closeModal={handleCloseModal}
+        />
+      )}
+      {openInfo && (
+        <CrashDumpModal
+          open={openInfo}
+          deviceType={deviceType}
+          onClose={handleClosCrashDumpModal}
+          onSubmit={handleDownloadCrashDump}
+        />
+      )}
     </>
   )
 }
 
 const mapStateToProps = (state: ReduxRootState) => ({
   hasCrashDump: Boolean(state.crashDump.data.files.length),
-  downloading: state.crashDump.status.downloading,
-  sending: state.crashDump.status.sending,
+  sending:
+    state.crashDump.loadingState === State.Loading ||
+    state.crashDump.downloadingState === State.Loading ||
+    state.crashDump.sendingState === State.Loading,
+  sended:
+    state.crashDump.downloadingState === State.Loaded &&
+    state.crashDump.sendingState === State.Loaded,
```

Comment:
No check for `state.crashDump.loadingState === State.Loaded` (not sure though if it would make sense)?

Reply:
@wojcik92michal in this case we don't want to check the loading state, because the backup can be already loaded